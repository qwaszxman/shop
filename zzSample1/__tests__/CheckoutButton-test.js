/*
 * CheckoutButton tests for JEST
 *
 * Copyright © Oleg Nosov 2016-Present
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE.txt file in the root directory of this source tree.
 */

import React from 'react';
import renderer from 'react-test-renderer';
import IntlMessageFormat from 'intl-messageformat';
import CheckoutButton
  from '../src/components/CheckoutButton/CheckoutButton';

const testCheckoutButtonLocalization = {
  checkoutTotal:
    'Checkout (Grand total {localizedCurrency}{total, plural, ' +
    '=0 {0}' +
    'other {#}})',
  GBP: '£',
};

const getLocalization = (id, params = {}) =>
  new IntlMessageFormat(
    testCheckoutButtonLocalization[id], 'en',
  ).format(params);

describe('CheckoutButton', () => {
  it('takes snapshot', () => {
    const renderedCheckoutButton = renderer.create(
      <CheckoutButton
        getLocalization={
          getLocalization
        }
        grandTotal={100}
        hidden={false}
        checkoutURL="/to/checkout/"
        currency="GBP"
      />,
    );
    expect(renderedCheckoutButton.toJSON()).toMatchSnapshot();
  });
});
