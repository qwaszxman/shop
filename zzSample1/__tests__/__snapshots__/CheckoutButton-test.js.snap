// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutButton takes snapshot 1`] = `
<a
  className="btn btn-primary btn-block animated"
  href="/to/checkout/"
  role="button"
>
  <i
    className="fa fa-shopping-cart mx-1"
  />
  <span>
    Checkout (Grand total Â£100)
  </span>
</a>
`;
